### PR TITLE
Adding Solace PubSub Component Docs

### PR DESCRIPTION
**Pull request by TKTheTechie**
Signed-off-by: TKTheTechie <thomas.kunnumpurath@solace.com>
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Adding documentation for a Solace PubSub+ AMQP Component

## Issue reference
https://github.com/dapr/docs/issues/3022

### CODE REVIEW

**Review by msfussell — APPROVED** [review]
LGTM! Thanks for contributing these docs for the component

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+title: "Solace-AMQP"
+linkTitle: "Solace-AMQP"
+description: "Detailed documentation on the Solace-AMQP pubsub component"
+aliases:
+  - "/operations/components/setup-pubsub/supported-pubsub/setup-solace-amqp/"
+---
+
+## Component format
+
+To setup Solace-AMQP pubsub create a component of type `pubsub.solace.amqp`. See [this guide]({{< ref "howto-publish-subscribe.md#step-1-setup-the-pubsub-component" >}}) on how to create and apply a pubsub configuration
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md]:
```suggestion
To setup Solace-AMQP pub/sub, create a component of type `pubsub.solace.amqp`. See [this guide]({{< ref "howto-publish-subscribe.md#step-1-setup-the-pubsub-component" >}}) on how to create and apply a pub/sub configuration.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+{{% alert title="Warning" color="warning" %}}
+The above example uses secrets as plain strings. It is recommended to use a secret store for the secrets as described [here]({{< ref component-secrets.md >}}).
+{{% /alert %}}
+
+## Spec metadata fields
+
+| Field              | Required | Details | Example |
+|--------------------|:--------:|---------|---------|
+| url    | Y  | Address of the AMQP broker. Can be `secretKeyRef` to use a secret reference. <br> Use the **`amqp://`** URI scheme for non-TLS communication. <br> Use the **`amqps://`** URI scheme for TLS communication. | `"amqp://host.domain[:port]"`
+| username | Y | The user name to connect to the broker with. Only required if anonymous is not specified or set to false | default
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md]:
```suggestion
| username | Y | The username to connect to the broker. Only required if anonymous is not specified or set to `false` .| `default`
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+| password | Y | The password to connect to the broker with. Only required if anonymous is not specified or set to false. | default
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md]:
```suggestion
| password | Y | The password to connect to the broker. Only required if anonymous is not specified or set to `false`. | `default`
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+| anonymous | N | To connect to the broker without credential validation. Only works if enabled on the broker. A username and password would not be required if this is set to true.| true
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md]:
```suggestion
| anonymous | N | To connect to the broker without credential validation. Only works if enabled on the broker. A username and password would not be required if this is set to `true`. | `true`
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+| caCert | Required for using TLS | Certificate Authority (CA) certificate in PEM format for verifying server TLS certificates. | `"-----BEGIN CERTIFICATE-----\n<base64-encoded DER>\n-----END CERTIFICATE-----"`
+| clientCert  | Required for using TLS | TLS client certificate in PEM format. Must be used with `clientKey`. | `"-----BEGIN CERTIFICATE-----\n<base64-encoded DER>\n-----END CERTIFICATE-----"`
+| clientKey | Required for using TLS | TLS client key in PEM format. Must be used with `clientCert`. Can be `secretKeyRef` to use a secret reference. | `"-----BEGIN RSA PRIVATE KEY-----\n<base64-encoded PKCS8>\n-----END RSA PRIVATE KEY-----"`
+
+### Communication using TLS
+
+To configure communication using TLS, ensure that the Solace broker  is configured to support certificates and provide the `caCert`, `clientCert`, `clientKey` metadata in the component configuration. For example:
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md]:
```suggestion
To configure communication using TLS:

1. Ensure that the Solace broker is configured to support certificates.
1. Provide the `caCert`, `clientCert`, and `clientKey` metadata in the component configuration. 

For example:
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    value: ${{ myLoadedClientCert }}
+  - name: clientKey
+    secretKeyRef:
+      name: mySolaceClientKey
+      key: mySolaceClientKey
+auth:
+  secretStore: <SECRET_STORE_NAME>
+```
+
+Note that while the `caCert` and `clientCert` values may not be secrets, they can be referenced from a Dapr secret store as well for convenience.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md]:
```suggestion
> While the `caCert` and `clientCert` values may not be secrets, they can be referenced from a Dapr secret store as well for convenience.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+### Publishing/subscribing to topics and queues
+
+By default, messages are published and subscribed over topics. If you would like your destination to be a queue - prefix the topic with `queue:` and the Solace AMQP component will connect to a queue.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md]:
```suggestion
By default, messages are published and subscribed over topics. If you would like your destination to be a queue, prefix the topic with `queue:` and the Solace AMQP component will connect to a queue.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+```bash
+docker run -d -p 8080:8080 -p 55554:55555 -p 8008:8008 -p 1883:1883 -p 8000:8000 -p 5672:5672 -p 9000:9000 -p 2222:2222 --shm-size=2g --env username_admin_globalaccesslevel=admin --env username_admin_password=admin --name=solace solace/solace-pubsub-standard
+```
+
+You can then interact with the server using the client port: `mqtt://localhost:5672`
+{{% /codetab %}}
+
+{{% codetab %}}
+You can also sign up for a free SaaS broker on [Solace Cloud](https://console.solace.cloud/login/new-account?product=event-streaming)
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md]:
```suggestion
You can also sign up for a free SaaS broker on [Solace Cloud](https://console.solace.cloud/login/new-account?product=event-streaming).
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+{{% /codetab %}}
+
+{{< /tabs >}}
+
+## Related links
+
+- [Basic schema for a Dapr component]({{< ref component-schema >}})
+- Read [this guide]({{< ref "howto-publish-subscribe.md#step-2-publish-a-topic" >}}) for instructions on configuring pub/sub components
+- [Pub/Sub building block]({{< ref pubsub >}})
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md]:
```suggestion
- [Pub/sub building block]({{< ref pubsub >}})
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
quick grammar pass

**Review by hhunter-ms — APPROVED** [review]

**Pull request merged**